### ISSUE TITLE
Potential for multiple Access-Control-Allow-Origin values to be set

### ISSUE THREAD

**Issue by joshuabaker** [craftcms/cms#6355]
Adding `Access-Control-Allow-Origin` can result in having multiple values (e.g. `domain1.com, domain2.com`), which will not be accepted by browsers.

https://github.com/craftcms/cms/blob/5eb272cea78758b267a113df85a544ec6e5ffa2b/src/controllers/GraphqlController.php#L87

https://github.com/craftcms/cms/blob/5eb272cea78758b267a113df85a544ec6e5ffa2b/src/controllers/GraphqlController.php#L93

I would suggest checking/merging and using `set`, instead of `add`. Or even `setDefault`.

**Comment by brandonkelly** [comment]
Those two lines are in separate logical branches, so they can’t both be run on the same request.

**Comment by joshuabaker** [comment]
@brandonkelly Please can we reopen this. I know it’s minor and I realise that it’s impossible for these both to run within the context of the `GraphqlController::actionApi` method, but the `add` means that you’re adding to any value set elsewhere. 

`Access-Control-Allow-Origin` should only ever be set to a single value to be accepted by browsers. By using `add` you’re making it impossible to set this elsewhere in the application for any purpose (and call the GraphQL action) without creating generally hard to diagnose CORS issues.

#### Simplified example scenario

A controller that I’ve made that sets the headers in `beforeAction` so that all actions are covered. The GraphQL action [dynamically adjusts the schema based on the current user group](https://github.com/craftcms/cms/issues/5416#issuecomment-572766918) but also allows for other authentication actions.

```php
class ApiController extends Controller
{
    public function beforeAction($action): bool
    {
        // Setting headers in here for all controller actions

        return parent::beforeAction($action);
    }

    public function actionUser(): ?Response
    {
        // Gets and returns logged in user state
    }

    public function actionGraphql(): ?Response
    {
        // Switching the schema based on the logged in user’s attributes

        return Craft::$app->runAction('graphql/api');
    }
}
```

Obviously fixable using conditions or remembering to make a method call in each action. That somewhat defeats the purpose of `beforeAction` here though.

**Comment by brandonkelly** [comment]
Gotcha, I didn’t realize you were setting the header yourself elsewhere.

**Comment by joshuabaker** [comment]
Yeah, sorry, bad issue description on my part. Thanks for reopening. 👍
